core-bundle/tests/Controller/ContentElement/ContentElementControllerTest.php

```diff
@@ -228,11 +235,36 @@ public function testDoesNotSetTheSharedMaxAgeIfTheElementHasNeitherAStartNorASto
         $controller = new TestSharedMaxAgeController();
         $controller->setContainer($container);
 
-        $response = $controller(new Request(), new ContentModel(), 'main');
+        $response = $controller(new Request(), $this->getContentModel(), 'main');
 
         $this->assertNull($response->getMaxAge());
     }
 
+    public function testUsesFragmentTemplateForSubrequests(): void
+    {
+        $framework = $this->mockContaoFramework();
+        $framework
+            ->expects($this->once())
+            ->method('createInstance')
+            ->with(FragmentTemplate::class, ['ce_test'])
+            ->willReturn(new FragmentTemplate('ce_test'))
+        ;
+
+        $this->container->set('contao.framework', $framework);
+        $this->container->set('contao.routing.scope_matcher', $this->mockScopeMatcher());
+
+        $currentRequest = new Request([], [], ['_scope' => 'frontend']);
+        $requestStack = $this->container->get('request_stack');
+
+        $requestStack->push(new Request()); // Main request
+        $requestStack->push($currentRequest); // Sub request
+
+        $controller = new TestController();
+        $controller->setContainer($this->container);
+
+        $controller(new Request(), $this->getContentModel(), 'main');
```

Comment:
Should this not use `$currentRequest`?

```suggestion
        $controller($currentRequest, $this->getContentModel(), 'main');
```

Reply:
Hah, yes it was meant to. But the test shows it works anyway, because the given one is not the main request 😅